clang/lib/CConv/ConstraintVariables.cpp

```diff
@@ -1323,6 +1310,43 @@ void PointerVariableConstraint::brainTransplant(ConstraintVariable *FromCV) {
   }
 }
 
+void PointerVariableConstraint::mergeDeclaration(ConstraintVariable *FromCV) {
```

Comment:
I'm confused about this code. I would have thought that `FromCV` is the new declaration that you just reached, while `this` is the one you've been using in the program up until now. As such, the `argumentConstraints` of `FromCV` should be empty, and you should not overwrite the `this->argumentConstraints` you already have.

Likewise, if you already have a `this-iTypeStr` wouldn't you want to keep it, and merge it with that of `FromCV` ?

Reply:
Yes you are right, `argumentConstraints` should not be rewritten. However, the `iTypeStr` will be rewritten only when the new declaration also has itype

Reply:
But don't we need to merge? That is, what if the first decl is
`int *foo(int *x) : itype(_Ptr<int>); // on the return`
but then the second is
`int *foo(int *x : itype(_Ptr<int>)); // on the arg`
Don't we want to merge these two, so that effectively the combined prototype is
`int *foo(int *x : itype(_Ptr<int>)) : itype(_Ptr<int>); // on both`
?

Reply:
They will be merged because each Pointer will have its own ConstraintVariable.

Reply:
OK, I'm happy to take your word for it! :)